src/fsharp/service/ServiceNavigation.fs

```diff
@@ -98,15 +98,18 @@ module NavigationImpl =
     /// Get information for implementation file      
     let getNavigationFromImplFile (modules: SynModuleOrNamespace list) =
         // Map for dealing with name conflicts
-        let mutable nameMap = Map.empty 
+        let names = System.Collections.Generic.Dictionary<_,_>()
```

Comment:
Could you open the namespace please? It also doesn't need `<_,_>` here.

Reply:
Doesn't that namespace conflict with `List`? But you can use `dict` perhaps, it's a type alias for dictionary.

Reply:
Nope, List module works fine, just as String module and String types can be used together when System is opened.

Reply:
@auduchinok, you're right, that's usually true, unless, for instance, you declare a list explicitly like `List<'T>`. Thanks to your comment I could clean up some old code myself ;).

Reply:
@abelbraaksma Could you share an example of where it wouldn't work please?

Reply:
@auduchinok, sorry, I forgot about this thread, but simply put, if you have a `let f (a: List<_>) = ...`, without `open System.Collections.Generic`, this will resolve to F#'s `list`, if you add the `open` statement, it'll resolve to the BCL list.

If you always use postfix notation like `let f (a: list) = ...` instead of `List<_>` or `List<'T>` the problem won't occur.

Reply:
@auduchinok, here's a repro, just paste this in a new project and you'll get errors. Remove the `open` and it'll compile:

```f#
open System.Collections.Generic

[<EntryPoint>]
let main argv =
    let f (x: List<_>) = List.sum x
    let a = f [1;2;3]
    0
```

Reply:
@abelbraaksma Thanks, it indeed reproduces! I have somehow didn't get into this issue before despite using both BCL collections and F# lists. Probably it's because I usually use them in separate places. 🙂